apps/sample-ui/src/mocks/handlers.ts

```diff
@@ -1,28 +1,33 @@
 import { rest } from 'msw'
+import { recordRestResponse } from './resolver'
 
 export const handlers = [
+  rest.get(`${process.env.REACT_APP_API_URL}/*`, recordRestResponse),
+  rest.post(`${process.env.REACT_APP_API_URL}/*`, recordRestResponse),
+  rest.patch(`${process.env.REACT_APP_API_URL}/*`, recordRestResponse),
+  rest.delete(`${process.env.REACT_APP_API_URL}/*`, recordRestResponse),
+
+  // custom interceptor
   rest.post(`${process.env.REACT_APP_API_URL}/login`, async (req, res, ctx) => {
     // Persist user's authentication in the session
     sessionStorage.setItem('is-authenticated', 'true')
-    const originalResponse = await ctx.fetch(req)
-    const originalResponseData = await originalResponse.json()
-    console.log(
-      '[msw] login handler: originalResponseData:',
-      originalResponseData
-    )
-
-    return res(
-      ctx.status(200),
-      ctx.json({
-        user: {
-          username: 'mockUserA',
-          permissions: ['AMAZING_ADMIN_PERMISSION'],
-        },
-      })
-    )
+    try {
+      const mock = require('./json/post-login.json')
+      return res(ctx.status(200), ctx.json(mock))
+    } catch (error) {
+      console.error(error)
+    }
   }),
 
-  rest.get(`${process.env.REACT_APP_API_URL}/magicitems`, (req, res, ctx) => {
-    return res(ctx.status(200), ctx.json(['mock response']))
-  }),
+  rest.get(
+    `${process.env.REACT_APP_API_URL}/magicitems`,
+    async (req, res, ctx) => {
+      try {
+        const mock = require('./json/get-magicitems.json')
+        return res(ctx.status(200), ctx.json(mock))
+      } catch (error) {
+        console.error(error)
+      }
+    }
+  ),
```

Comment:
After taking a look at the actual POC, I think hijacking the response might not be required, if we let the users use it themselves in cy.intercept. 

What bleublanc can help here maybe is to provide the code snippet for cy.intercept with the correct mock path for all the APIs that are recorded.

Reply:
> What bleublanc can help here maybe is to provide the code snippet for cy.intercept with the correct mock path for all the APIs that are recorded.

Agreed!